### PR TITLE
client: make -bench and -test able to handle any bias type

### PR DESCRIPTION
**Pull request by solaslin**
- Supported more bias types other than To and Talpha: made Tbias a template as well.
- Let hipblaslt-bench and -test follow the same logic of default bias type as ContractionProblem 
- Updated documentations

### CODE REVIEW

File: clients/gtest/matmul_gtest.yaml [hunk 1]
```diff
@@ -544,7 +543,7 @@ Tests:
   scaleA: [ 0, 1]
   scaleB: [ 0, 1]
   bias_vector: [0, 1]
-  bias_type: f32_r
+  bias_type: bf16_r
   unit_check: 1
```
Comment (solaslin) [inline clients/gtest/matmul_gtest.yaml:547]: The test cases can test for bias type != To or Talpha. And this is also the default bias type for F8SS gemm

File: clients/gtest/matmul_gtest.yaml [hunk 2]
```diff
@@ -622,9 +621,8 @@ Tests:
   scaleB: [0, 1]
   scaleC: [0, 1]
   scaleD: [0, 1]
-  # TODO- support bias types other than Tscale or To
-  # bias_vector: [0, 1]
-  # bias_type: f16_r
+  bias_vector: [0, 1]
+  bias_type: f16_r
   unit_check: 1
```
Comment (solaslin) [inline clients/gtest/matmul_gtest.yaml:626]: The test cases can test for bias type != To or Talpha. And this is also the default bias type for F8F8S gemm

File: clients/include/testing_matmul.hpp [hunk 3]
```diff
@@ -330,42 +317,89 @@ void check(hipStream_t                          stream,
...
+{
+    static const std::set<hipDataType> supported_bias_types
+        = {HIP_R_32F, HIP_R_64F, HIP_R_16F, HIP_R_16BF, HIP_R_32I};
+
+    hipDataType real_bias_type = arg.bias_type;
+    
+    // when bias type is unset.
+    if(arg.bias_type == HIPBLASLT_DATATYPE_INVALID)
+    {
+        if(arg.compute_type == HIPBLAS_COMPUTE_32I || arg.compute_type == HIPBLAS_COMPUTE_32F_FAST_TF32)
```
Comment (solaslin) [inline clients/include/testing_matmul.hpp:347]: This function follows the logic of RocblasltContractionProblem() in tensile_host.hpp.

File: clients/include/testing_matmul.hpp [hunk 4]
```diff
@@ -330,42 +317,89 @@ void check(hipStream_t                          stream,
...
+    }
+    else // if(real_bias_type == HIP_R_64F)
+    {
+        return testing_matmul_with_bias<TiA, TiB, To, Tc, Tci, double>(arg);
+    }
+}
+
+template <typename TiA, typename TiB, typename To, typename Tc, typename Tci, typename Tbias>
+void testing_matmul_with_bias(const Arguments& arg)
 {
```
Comment (solaslin) [inline clients/include/testing_matmul.hpp:404]: Make Tbias a template arg as well. Thus we can get rid of those if/else expressions switching between Talpha and To.

File: clients/include/testing_matmul.hpp [hunk 5]
```diff
@@ -330,42 +317,90 @@ void check(hipStream_t                          stream,
...
+    hipDataType real_bias_type = arg.bias_type;
+
+    // when bias type is unset.
+    if(arg.bias_type == HIPBLASLT_DATATYPE_INVALID)
+    {
+        if(arg.compute_type == HIPBLAS_COMPUTE_32I || arg.compute_type == HIPBLAS_COMPUTE_32F_FAST_TF32)
+        {
+            real_bias_type = HIP_R_32F;
+        }
+        else if((arg.a_type == HIP_R_8F_E4M3_FNUZ || arg.a_type == HIP_R_8F_E5M2_FNUZ)
```
Comment (solaslin) [inline clients/include/testing_matmul.hpp:352]: This function follows the logic of RocblasltContractionProblem() in tensile_host.hpp

File: clients/gtest/matmul_gtest.cpp [hunk 6]
```diff
@@ -123,8 +123,7 @@ namespace
                 if(arg.bias_vector)
                 {
                     name << "_BIAS" << hipblaslt_bias_source_to_string(arg.bias_source);
-                    if(arg.d_type != arg.scale_type && arg.bias_type == arg.scale_type)
-                        name << hip_datatype_to_string(arg.bias_type);
+                    name << "_" << hip_datatype_to_string(arg.bias_type);
                 }
```
Comment (solaslin) [inline clients/gtest/matmul_gtest.cpp:127]: coded the bias type anyway

**Review by KKyang — APPROVED** [review]

File: docs/functions.rst [hunk 7]
```diff
@@ -83,6 +83,64 @@ HIPBLAS_COMPUTE_32F     HIP_R_32F           HIP_R_16F       HIP_R_32F
...
+
++-------+-------+-------+-------+-------------+----------+----------+------------+-----------+
+| Atype | Btype | Ctype | Dtype | computeType | scaleA,B | scaleC,D | Bias Type  | Default   |
+|       |       |       |       |             |          |          |            | Bias Type |
++=======+=======+=======+=======+=============+==========+==========+======+=====+===========+
+| FP8   | FP8   | FP16  | FP16  | FP32        | Yes      | No       | FP32, FP16 | FP16      |
+|       |       +-------+-------+             +          +          +------------+-----------+
+|       |       | BF16  | BF16  |             |          |          | FP32, BF16 | BF16      |
+|       |       +-------+-------+             +          +          +------------+-----------+
+|       |       | FP32  | FP32  |             |          |          | FP32, BF16 | BF16      |
```
Comment (solaslin) [inline docs/functions.rst:107]: Added the current status of FP8 matmul type combination

**Review by jichangjichang — APPROVED** [review]
lgtm

**Pull request merged**